### PR TITLE
[MM-40952] migrations: Add if statements for some queries

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
- Added new semantic version checking inside the scripts.
- Add if statements where applicable.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-40952

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: db/migrations/mysql/000051_create_msg_root_count.up.sql [hunk 1]
```diff
@@ -14,8 +14,8 @@ SET @tmp_count_root := 0;
 	IF(MentionCountRoot_EXIST = 1) THEN
 		SELECT @tmp_count_root := COUNT(*)
 		FROM ChannelMembers
-			WHERE MsgCountRoot = NULL
-			OR MentionCountRoot = NULL;
+			WHERE MsgCountRoot IS NULL
```
Comment (agnivade) [inline db/migrations/mysql/000051_create_msg_root_count.up.sql:17]: Is there a difference between the two?

**Review by agnivade — APPROVED** [review]
Great work!

**Comment by agnivade** [comment]
Sorry, forgot one thing: do we want to take care of this?

```
IF(ParentId_EXIST > 0) THEN
    UPDATE Posts SET RootId = ParentId WHERE RootId = '' AND RootId != ParentId;
    ALTER TABLE Posts MODIFY COLUMN FileIds text, MODIFY COLUMN Props JSON, DROP COLUMN ParentId;
ELSE
    ALTER TABLE Posts MODIFY COLUMN FileIds text, MODIFY COLUMN Props JSON;
END IF;
END;
```

**Comment by isacikgoz** [comment]
Good point @agnivade I added some checks to the both mysql/postgres scripts. PTAL.

File: db/migrations/mysql/000051_create_msg_root_count.up.sql (hunk 1 above)
Comment (isacikgoz) [reply db/migrations/mysql/000051_create_msg_root_count.up.sql:17]: Yeah, it seems `NULL` values are not comparable with `=, <, or <>`. Also updated some other occurrences in [79e6b57](https://github.com/mattermost/mattermost-server/pull/19359/commits/70e6b5741af9fd39de45c965c682c4d16e2c3071).

File: db/migrations/mysql/000059_upgrade_users_v6.0.down.sql [hunk 2]
```diff
@@ -36,16 +36,37 @@ SET @preparedStatement = (SELECT IF(
         WHERE table_name = 'Users'
         AND table_schema = DATABASE()
         AND column_name = 'Timezone'
-        AND column_default = NULL
+        AND column_type = 'JSON'
```
Comment (agnivade) [inline db/migrations/mysql/000059_upgrade_users_v6.0.down.sql:39]: Isn't this same as the previous one? `!= 'varchar(256)'` would mean anything right?

File: db/migrations/mysql/000059_upgrade_users_v6.0.down.sql [hunk 3]
```diff
@@ -36,16 +36,37 @@ SET @preparedStatement = (SELECT IF(
...
+    'SELECT 1'
+));
+
+PREPARE alterIfExists FROM @preparedStatement;
+EXECUTE alterIfExists;
+DEALLOCATE PREPARE alterIfExists;
+
+PREPARE alterIfExists FROM @preparedStatement;
+EXECUTE alterIfExists;
+DEALLOCATE PREPARE alterIfExists;
```
Comment (agnivade) [inline db/migrations/mysql/000059_upgrade_users_v6.0.down.sql:68]: Unintentional repetition?

Comment (isacikgoz) [reply db/migrations/mysql/000059_upgrade_users_v6.0.down.sql:68]: Yeah, nice catch. I need a linter for sql files 🤦

File: db/migrations/mysql/000059_upgrade_users_v6.0.down.sql (hunk 2 above)
Comment (isacikgoz) [reply db/migrations/mysql/000059_upgrade_users_v6.0.down.sql:39]: Hmm, actually we can use either. 0/5 checking if data type is `JSON`. This is for purely reverting the `up` script. Actually both would do the same in this context. And there are two prepared statements for this column; one for column_type the other one is column_default.

Comment (agnivade) [reply db/migrations/mysql/000059_upgrade_users_v6.0.down.sql:39]: Ok, let's add a comment if we are keeping it unchanged. :+1:

Comment (isacikgoz) [reply db/migrations/mysql/000059_upgrade_users_v6.0.down.sql:39]: Oh man, I'm removing this. I need more coffee :) You're right the previous statement is enough.

Comment (agnivade) [reply db/migrations/mysql/000059_upgrade_users_v6.0.down.sql]: No worries man. You need a break. :)

**Review by agnivade — APPROVED** [review]

**Comment by agnivade** [comment]
@amyblais - We would like this to be in the upcoming 6.4 release. Should this be cherry-picked to cloud?

**Comment by isacikgoz** [comment]
@amyblais I think it would be great if we cherry-pick this to cloud, WDYT?

**Comment by agnivade** [comment]
lol, our responses crossed :facepalm:

**Comment by amyblais** [comment]
Currently v6.4 will be based off next week's cloud release, I'll own cherry-picking this if needed.

**Pull request merged**